consensus/ethash/consensus.go

```diff
@@ -17,28 +17,68 @@
 package ethash
 
 import (
+	"fmt"
 	"math/big"
 
 	"github.com/Taraxa-project/taraxa-evm/common"
 	"github.com/Taraxa-project/taraxa-evm/core/types"
 	"github.com/Taraxa-project/taraxa-evm/core/vm"
+	"github.com/Taraxa-project/taraxa-evm/taraxa/state/rewards_stats"
 )
 
 var (
-	FrontierBlockReward       = big.NewInt(5e+18) // Block reward in wei for successfully mining a block
-	ByzantiumBlockReward      = big.NewInt(3e+18) // Block reward in wei for successfully mining a block upward from Byzantium
-	ConstantinopleBlockReward = big.NewInt(2e+18) // Block reward in wei for successfully mining a block upward from Constantinople
+	TaraPrecision   = big.NewInt(1e+18)              // Tara precision
+	YieldPercentage = big.NewInt(20)                 // 20% yield
+	BlocksPerYear   = big.NewInt(365 * 24 * 60 * 15) // 365 days * 24 hours * 60 minutes * 15 (1 pbft block every 4 seconds -> 15 per minute)
 )
 
+type BlockNumAndCoinbase = struct {
+	Number types.BlockNum
+	Author common.Address
+}
+
 // Some weird constants to avoid constant memory allocs for them.
 var (
 	big8  = big.NewInt(8)
 	big32 = big.NewInt(32)
 )
 
-type BlockNumAndCoinbase = struct {
-	Number types.BlockNum
-	Author common.Address
+var (
+	FrontierBlockReward       = big.NewInt(5e+18) // Block reward in wei for successfully mining a block
+	ByzantiumBlockReward      = big.NewInt(3e+18) // Block reward in wei for successfully mining a block upward from Byzantium
+	ConstantinopleBlockReward = big.NewInt(2e+18) // Block reward in wei for successfully mining a block upward from Constantinople
+)
+
+func AccumulateRewardsDpos(totalStake *big.Int, rewardsStats *rewards_stats.RewardsStats, rewardsPayouts map[common.Address]*big.Int) {
+	// Calculates number of tokens to be generated as block reward
+	blockReward := new(big.Int).Mul(totalStake, YieldPercentage)
+	blockReward.Div(blockReward, big.NewInt(100))
```

Comment:
I think this is const value `bigutil.Big100` or something like that

Reply:
tried, didnt find this constatnt in that package...

Reply:
hmm, you need to add it to index.go as are others `var Big32 = cache[32]` so something like `var Big100 = cache[100]`